scipy/stats/_morestats.py

```diff
@@ -1311,28 +1339,33 @@ def _all(x):
         message = ("The `optimizer` argument of `boxcox_normmax` must return "
                    "an object containing the optimal `lmbda` in attribute `x`.")
         raise ValueError(message)
-    else:
-        # Test if the optimal lambda causes overflow
-        x = np.asarray(x)
-        x_treme = np.max(x, axis=0) if np.any(res > 0) else np.min(x, axis=0)
-        istransinf = np.isinf(special.boxcox(x_treme, res))
-        dtype = x.dtype if np.issubdtype(x.dtype, np.floating) else np.float64
-        if np.any(istransinf):
-            warnings.warn(
+    elif not np.isinf(ymax):  # adjust the final lambda
+        # x > 1, boxcox(x) > 0; x < 1, boxcox(x) < 0
+        xmax, xmin = np.max(x), np.min(x)
+        if xmin >= 1:
+            x_treme = xmax
+        elif xmax <= 1:
+            x_treme = xmin
+        else:  # xmin < 1 < xmax
+            indicator = special.boxcox(xmax, res) > abs(special.boxcox(xmin, res))
+            x_treme = xmax if np.any(indicator) else xmin
```

Comment:
You use `any` because `res` is an array (when `method='all'`), right?
Since we are working on the `method='mle'` part of this, then would it be more correct to choose `x_treme` based on the element of `res` that corresponds with MLE?